src/input.rs

```diff
@@ -603,6 +603,20 @@ impl<'a, A: ActionContext + 'a> Processor<'a, A> {
     }
 
     pub fn on_mouse_wheel(&mut self, delta: MouseScrollDelta, phase: TouchPhase, modifiers: ModifiersState) {
+        // Make sure the new and deprecated setting are both allowed
+        let faux_scrolling_lines = self.mouse_config
+            .faux_scrollback_lines
+            .unwrap_or(self.scrolling_config.faux_multiplier as usize);
+
+        let use_faux_scrolling = self.ctx.terminal_mode().contains(TermMode::ALT_SCREEN)
+            && faux_scrolling_lines > 0 && !modifiers.shift;
+
+        let scrolling_multiplier = if use_faux_scrolling {
+            faux_scrolling_lines
+        } else {
+            self.scrolling_config.multiplier as usize
+        };
```

Comment:
Faux scrolling and normal scrolling are two completely different things. It's not possible to check just for the presence of one and use the other if it's not present.

The faux scrolling multiplier should be used for faux scrolling and the scrolling multiplier should be used for normal scrolling.

Reply:
Hi,

In order to check which multiplier should be used, I moved that code that from `scroll_terminal` to `on_mouse_wheel`, and made `scroll_terminal` always scrolls `scroll_multiplier` lines, hence the ugly code :(
Is there a better way to do that?